app/src/main/java/gov/hhs/cdc/trustedintermediary/external/javalin/DomainsRegistration.java

```diff
@@ -66,6 +73,28 @@ static void registerDomainsHandlers(Javalin app, Set<DomainConnector> domains) {
                                         }));
     }
 
+    static void registerOpenApi(Javalin app, Set<DomainConnector> domains) {
+        Set<String> openApiSpecifications =
+                domains.stream()
+                        .map(DomainConnector::openApiSpecification)
+                        .collect(Collectors.toSet());
+
+        // not using @Inject in a field of this class because we are still bootstrapping the
+        // application context
+        // also not using a static field because we need to register different YamlCombiners in the
+        // unit tests
+        String fullOpenApiSpecification =
+                ApplicationContext.getImplementation(OpenApi.class)
+                        .generateApiDocumentation(openApiSpecifications);
+
+        app.get(
+                "/openapi",
+                ctx -> {
+                    ctx.header("Content-Type", "application/yaml");
```

Comment:
Should we add logging when the endpoint is accessed?

Reply:
Good idea!